superduperdb/ext/llm/base.py

```diff
@@ -0,0 +1,317 @@
+import abc
+import asyncio
+import dataclasses as dc
+import inspect
+import typing
+from functools import reduce, wraps
+from logging import WARNING, getLogger
+from typing import Any, Callable, List, Optional, Union
+
+from superduperdb import logging
+from superduperdb.components.component import Component
+from superduperdb.components.model import _Predictor
+from superduperdb.ext.utils import format_prompt
+
+if typing.TYPE_CHECKING:
+    from superduperdb.base.datalayer import Datalayer
+
+# Disable httpx info level logging
+getLogger("httpx").setLevel(WARNING)
+
+
+def ensure_initialized(func):
```

Comment:
Maybe we need this as a general purpose tool (not just for LLMs)?

Reply:
Yes, But it is only used for LLMs now. Because in the previous design, the model was instantiated first and then passed to SuperDuperDB.
We can make it as a general tool in this plan https://github.com/SuperDuperDB/superduperdb/discussions/1604